test/unit/detect_stale_deprecated_references_test.rb

```diff
@@ -0,0 +1,47 @@
+# typed: false
+# frozen_string_literal: true
+
+require "test_helper"
+require "packwerk/detect_stale_deprecated_references"
+require "byebug"
+
+module Packwerk
+  class DetectStaleDeprecatedReferencesTest < Minitest::Test
```

Comment:
Ditto my comment about the other unit tests: after `DeprecatedReferences#stale_violations?` is added, these tests could go in their own commit along with the `DetectStaleDeprecatedReferences#stale_violations?` implementation they’re testing, and then a third commit could finally implement `Cli#detect_stale_violations`. As well as making it clearer which changes go together, this sequence of commits would make it clearer which pieces build on top of which other pieces.

Reply:
I went ahead and did this. let me know if you have any concerns